src/SignalR/server/StackExchangeRedis/src/Internal/RedisProtocol.cs

```diff
@@ -80,21 +82,23 @@ public byte[] WriteGroupCommand(RedisGroupCommand command)
             // * A 'str': The connection Id
             // Any additional items are discarded.
 
-            var writer = MemoryBufferWriter.Get();
+            var memoryBufferWriter = MemoryBufferWriter.Get();
             try
             {
-                MessagePackBinary.WriteArrayHeader(writer, 5);
-                MessagePackBinary.WriteInt32(writer, command.Id);
-                MessagePackBinary.WriteString(writer, command.ServerName);
-                MessagePackBinary.WriteByte(writer, (byte)command.Action);
-                MessagePackBinary.WriteString(writer, command.GroupName);
-                MessagePackBinary.WriteString(writer, command.ConnectionId);
-
-                return writer.ToArray();
+                var writer = new MessagePackWriter(memoryBufferWriter);
+
+                writer.WriteArrayHeader(5);
+                writer.WriteInt32(command.Id);
```

Comment:
`writer.Write` would be better to use than `writer.WriteInt32`. That should preserve the old semantic of writing the most compressed version of the integer. Calling `MessagePackWriter.WriteInt32` will force it to use 5 bytes of msgpack even for small integers.

Reply:
Unfortunately I don't think we can change that right now, old clients/old servers will be expecting an int32 and if a new client/new server writes it as an int16 or int8 then the old clients/old servers will fail to parse the payload.

Reply:
Should we update the version of the "MessagePackProtocol" then ?

Reply:
No, I don't think it's worth it to save a few bytes and it's a bit more complicated than just updating the version of the protocol 😄 

The scope of this PR should be limited to updating the library version and not introduce any differences in the bytes written to the wire.

Reply:
noted

Reply:
No, my point is that the old messagepack was *already* writing compressed integers when you call `MessagePackBinary.WriteInt32`. You get the equivalent behavior in v2 by calling `MessagePackWriter.Write(int)`. 

If you call `MessagePackWriter.WriteInt32(int)` that's equivalent to calling v1.8's `MessagePackBinary.WriteInt32ForceInt32Block` which you weren't calling here before the change.

Reply:
Hmm, that's not the behavior I remember observing, but I could be wrong.

Reply:
Ya, you're wrong. :)

[This](https://github.com/neuecc/MessagePack-CSharp/blob/c7a4f10920aaea283106d2dc7a85a7f9038da25c/src/MessagePack/MessagePackBinary.cs#L1314) is the method you were calling before. See how the implementation checks the value of the integer and compresses the msgpack written out?

[This](https://github.com/neuecc/MessagePack-CSharp/blob/c7a4f10920aaea283106d2dc7a85a7f9038da25c/src/MessagePack/MessagePackBinary.cs#L1394) is the force 5 bytes behavior method from v1.8, which you were _not_ calling.

Now consider [v2.0's `Write(int)` method](https://github.com/neuecc/MessagePack-CSharp/blob/9ed868e36921eb27ce68400c3fa38f2a5e456a00/src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackWriter.cs#L399) which compresses and [v2.0's `WriteInt32(int)`](https://github.com/neuecc/MessagePack-CSharp/blob/9ed868e36921eb27ce68400c3fa38f2a5e456a00/src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackWriter.cs#L439) which does _not_ compress.

Therefore the equivalency table is:

| v1.x | v2.x |
|--|--|
| MessagePackBinary.WriteInt32 | MessagePackWriter.Write(int) |
| MessagePackBinary.WriteInt32ForceInt32Block | MessagePackWriter.WriteInt32 |

Reply:
🦆 
good to know (learning a lot during this PR)